src/bundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\TwigComponents\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder(IbexaTwigComponentsExtension::EXTENSION_NAME);
+        $rootNode = $treeBuilder->getRootNode();
+
+        $rootNode
+            ->useAttributeAsKey('group')
+            ->arrayPrototype()
+                ->useAttributeAsKey('name')
+                ->arrayPrototype()
+                    ->children()
+                        ->scalarNode('type')
+                            ->isRequired()
+                            ->validate()
+                                ->ifNotInArray(['script', 'stylesheet', 'template', 'controller', 'html'])
+                                ->thenInvalid('Invalid type "%s". Allowed types: script, stylesheet, template, controller, html')
```

Comment:
I would extract this list to a const for easier maintenance whenever a new type arrives.